wire-ios-data-model/Tests/UseCases/IsSelfUserE2EICertifiedUseCaseTests.swift

```diff
@@ -0,0 +1,169 @@
+        mockSafeCoreCrypto = nil
+
+        super.tearDown()
+    }
+
+    func testExpiredCertificateResultsInFalse() async throws {
+        // Given
+        mockSafeCoreCrypto.coreCrypto.getUserIdentitiesConversationIdUserIds_MockMethod = { conversationID, userIDs in
+            XCTAssertEqual(conversationID, .init(base64Encoded: "qE4EdglNFI53Cm4soIFZ/rUMVL4JfCgcE4eo86QVxSc=")!)
+            XCTAssertEqual(userIDs, ["36dfe52f-157d-452b-a9c1-98f7d9c1815d"]) // TODO [WPB-765]: after CC update it should have the suffix "@example.com"
+            return [userIDs[0]: [.withStatus(.valid), .withStatus(.expired)]]
+        }
+
+        // When
+        let isCertified = try await sut.invoke()
+
+        // Then
+        XCTAssertFalse(isCertified)
+    }
+
+    func testRevokedCertificateResultsInFalse() async throws {
+        // Given
+        mockSafeCoreCrypto.coreCrypto.getUserIdentitiesConversationIdUserIds_MockMethod = { _, userIDs in
+            [userIDs[0]: [.withStatus(.valid), .withStatus(.revoked)]]
+        }
+
+        // When
+        let isCertified = try await sut.invoke()
+
+        // Then
+        XCTAssertFalse(isCertified)
+    }
+
+    func testValidCertificatesResultsInTrue() async throws {
+        // Given
+        mockSafeCoreCrypto.coreCrypto.getUserIdentitiesConversationIdUserIds_MockMethod = { _, userIDs in
+            [userIDs[0]: [.withStatus(.valid), .withStatus(.valid)]]
+        }
+
+        // When
+        let isCertified = try await sut.invoke()
+
+        // Then
+        XCTAssertTrue(isCertified)
+    }
+
+    func testEmptyResultEvaluatesToFalse() async throws {
+        // Given
+        mockSafeCoreCrypto.coreCrypto.getUserIdentitiesConversationIdUserIds_MockMethod = { _, _ in
+            [:]
+        }
+
+        // When
+        let isCertified = try await sut.invoke()
+
+        // Then
+        XCTAssertFalse(isCertified)
+    }
+
+    func testEmptyIdentitiesEvaluatesToFalse() async throws {
+        // Given
+        mockSafeCoreCrypto.coreCrypto.getUserIdentitiesConversationIdUserIds_MockMethod = { _, userIDs in
+            [userIDs[0]: []]
+        }
+
+        // When
+        let isCertified = try await sut.invoke()
+
+        // Then
+        XCTAssertFalse(isCertified)
+    }
+
+    // MARK: - Helpers
+
+    private func setupUsersClientsAndConversation() {
+        context.performAndWait {
+            setupMLSSelfConversation()
+            setupSelfUser()
+            setupClients()
+        }
+    }
+
+    private func setupMLSSelfConversation() {
+        let conversation = ZMConversation.insertNewObject(in: context)
+        conversation.remoteIdentifier = .init(uuidString: "11AE029E-AFFA-4B81-9095-497797C0C0FA")
+        conversation.mlsGroupID = .init(base64Encoded: "qE4EdglNFI53Cm4soIFZ/rUMVL4JfCgcE4eo86QVxSc=")
+        conversation.messageProtocol = .mls
+        conversation.mlsStatus = .ready
+        conversation.conversationType = .`self`
+    }
```

Comment:
FYI in `DataModelSupport` there is a `ModelHelper` that contains code like this to set up core data objects in tests. Expanding that helper object will help prevent duplication.

Reply:
@caldrian please check this comment

Reply:
Opened a PR: https://github.com/wireapp/wire-ios/pull/1032